src/app/components/tx-item.tsx

```diff
@@ -288,9 +289,9 @@ export const TxItem = ({ transaction, ...rest }: TxItemProps) => {
       >
         <TxItemIcon transaction={transaction} />
         <SpaceBetween flexGrow={1}>
-          <Stack spacing="base-tight">
+          <Stack spacing="base-tight" whiteSpace="nowrap">
             <Title as="h3" fontWeight="normal">
-              {getTxTitle(transaction)}
+              {truncateText(getTxTitle(transaction), 20)}
```

Comment:
Wouldn't it be better if we truncate and show the ellipsis [purely with CSS](https://css-tricks.com/flexbox-truncated-text/)? I have reservations to remove characters as it makes the transaction less readable (by humans and screen readers). If we do it via CSS we truncate based on available white space: for ex. in the full tab view truncation will not be needed unless the screen width is reduced.

Additionally, we should add a tooltip on hover to the truncated elements only. The tooltip would that displays the full text.  

See the example below
![CleanShot 2022-05-05 at 18 56 26@2x](https://user-images.githubusercontent.com/239215/166975164-8b564cbb-ac6a-4d62-af40-368171b75f38.png)

Reply:
Yep, I can change to using CSS and add a tooltip, np.

Reply:
Will wait for the new changes as mentioned to be implemented prior to QA testing

> Yep, I can change to using CSS and add a tooltip, np.

Will wait for the new changes as mentioned to be implemented prior to QA testing